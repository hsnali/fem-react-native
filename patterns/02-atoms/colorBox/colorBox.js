import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

export const ColorBox = ({ name, code }) => {
  const textColor = {
    color:
      parseInt(code.replace('#', ''), 16) > 0xffffff / 1.1 ? '#222' : '#eee',
  };
  return (
    <View style={[styles.container, { backgroundColor: code }]}>
      <Text style={[styles.color, textColor]}>
        {name}: {code}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 25,
  },
  color: {
    fontWeight: '700',
    textAlign: 'center',
  },
});

export default ColorBox;
