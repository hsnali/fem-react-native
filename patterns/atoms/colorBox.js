import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

export const ColorBox = ({ name, color }) => {
  return (
    <View style={[styles.container, { backgroundColor: color }]}>
      <Text style={styles.color}>
        {name}: {color}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginVertical: 5,
    padding: 10,
    borderRadius: 2,
  },
  color: {
    color: '#ffffff',
    fontWeight: '700',
    textAlign: 'center',
  },
});

export default ColorBox;
