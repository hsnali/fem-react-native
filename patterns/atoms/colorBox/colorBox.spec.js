import React from 'react';
import { render, toJSON } from '@testing-library/react-native';
import { toHaveStyle } from '@testing-library/jest-native';
import { ColorBox } from './colorBox';

expect.extend({ toHaveStyle });

const props = {
  name: 'Red',
  code: '#F00',
};

let wrapper = render(<ColorBox {...props} />);

it('snapshot test', () => {
  const { container } = wrapper;

  expect(container).toMatchSnapshot();
});

it('renders color name', () => {
  const { getByText } = wrapper;
  const text = getByText(props.name, { exact: false });

  expect(text).toBeTruthy();
});

it('renders color value', () => {
  const { getByText } = wrapper;
  const text = getByText(props.code, { exact: false });

  expect(text).toBeTruthy();
});

it('sets background color', () => {
  const { container } = wrapper;
  const [root] = container.children;
  const style = {
    backgroundColor: props.code,
  };

  expect(root).toHaveStyle(style);
});
