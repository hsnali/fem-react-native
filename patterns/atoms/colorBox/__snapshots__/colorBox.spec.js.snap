// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 2,
          "marginVertical": 5,
          "padding": 10,
        },
        Object {
          "backgroundColor": "#F00",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontWeight": "700",
            "textAlign": "center",
          },
          Object {
            "color": "#eee",
          },
        ]
      }
    >
      Red
      : 
      #F00
    </Text>
  </View>
</View>
`;
